### PR TITLE
updated documentation for databricks operator

### LINKED ISSUE #23954: Add missing parameter documentation in `DatabricksSubmitRunOperator` and `DatabricksSqlOperator`
### Body

Currently the following modules are missing certain parameters in their docstrings. Because of this, these parameters are not captured in the [Python API docs for the Databricks provider](https://airflow.apache.org/docs/apache-airflow-providers-databricks/stable/_api/airflow/providers/databricks/index.html).

- [ ] DatabricksSubmitRunOperator: `tasks`
- [ ] DatabricksSqlOperator: `do_xcom_push`
   - Granted this is really part of the `BaseOperator`, but this operator specifically sets the default value to False so it would be good if this was explicitly listed for users.

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by Bowrna**
closes: #23954

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/databricks/operators/databricks.py [hunk 1]
```diff
@@ -206,6 +206,39 @@ class DatabricksSubmitRunOperator(BaseOperator):
...
+                            "notebook_path": "/Users/airflow@example.com/PrepareData"
+                    }
+            }
+        ]
+        notebook_run = DatabricksSubmitRunOperator(
+            task_id='notebook_run',
+            tasks = tasks)
+
+    :param tasks: Array of Objects(RunSubmitTaskSettings) <= 100 items.
+    The supported params in the array are
```
Comment (mik-laj) [inline airflow/providers/databricks/operators/databricks.py]: Can you add indentation here?

File: airflow/providers/databricks/operators/databricks.py [hunk 2]
```diff
@@ -206,6 +206,37 @@ class DatabricksSubmitRunOperator(BaseOperator):
...
+        - ``existing_cluster_id``
+        - ``new_cluster``
+        - ``notebook_task``
+        - ``spark_jar_task``
+        - ``spark_python_task``
+        - ``spark_submit_task``
+        - ``pipeline_task``
+        - ``python_wheel_task``
+        - ``libraries``
+        - ``timeout_seconds``
```
Comment (eladkal) [inline airflow/providers/databricks/operators/databricks.py]: No need to list the options the link to the API is enough.
We don't want to maintain this list everytime there is a new option in the API.

File: airflow/providers/databricks/operators/databricks.py [hunk 3]
```diff
@@ -150,7 +150,7 @@ class DatabricksSubmitRunOperator(BaseOperator):
     <https://docs.databricks.com/dev-tools/api/latest/jobs.html#operation/JobsRunsSubmit>`_
     API endpoint.
 
-    There are two ways to instantiate this operator.
+    There are three ways to instantiate this operator.
```
Comment (eladkal) [inline airflow/providers/databricks/operators/databricks.py:154]: This operator has 186 lines of text in docstring. That is a lot! I feel we should extract all of this information to howto guide of the operator and keep minimal description on the class code.
WDYT @josh-fell @Bowrna ?

Comment (Bowrna) [reply airflow/providers/databricks/operators/databricks.py:154]: yes it's better to move the information to howto guide. i will do it and update in this PR.

File: docs/apache-airflow-providers-databricks/operators/submit_run.rst [hunk 4]
```diff
@@ -52,6 +52,8 @@ one named parameter for each top level parameter in the ``runs/submit`` endpoint
 
 All other parameters are optional, and described in the documentation of the ``DatabricksSubmitRunOperator`` class.
 
+Another way to do is use the param tasks to pass array of objects to instantiate this operator. Here the value of tasks param that is used to invoke ``api/2.1/jobs/runs/submit`` endpoint is passed through the ``tasks`` param in ``DatabricksSubmitRunOperator``. Instead of invoking single task, you can pass array of task and submit a one-time run.
+
```
Comment (Bowrna) [inline docs/apache-airflow-providers-databricks/operators/submit_run.rst:103]: @eladkal @josh-fell This howto documentation already contains information about instantiating the operator. It's additionally present in docstring with example too. Could you tell me if we can move the example part from docstring and retain this information only in howto guide?

Comment (eladkal) [reply docs/apache-airflow-providers-databricks/operators/submit_run.rst:103]: In my opinion there is no need to duplicate information the operator can reference the docs (we do that in other operators as well)

Comment (potiuk) [reply docs/apache-airflow-providers-databricks/operators/submit_run.rst:103]: yes

**Comment by potiuk** [comment]
I guess it's close to be ready @Bowrna  ? just some doc failures :) ?

**Comment by Bowrna** [comment]
I have fixed it @potiuk :)

**Review by potiuk — APPROVED** [review]

**Pull request merged**